src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -237,7 +279,20 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 
 		$options = wp_parse_args( $options, array( 'with_supports' => true ) );
 
-		if ( null === static::$theme || ! static::has_same_registered_blocks( 'theme' ) ) {
+		$can_use_cached = ! wp_is_development_mode( 'theme' );
+		if ( $can_use_cached ) {
+			$cache_key = 'get_theme_data_cache_' . md5( wp_json_encode( $options ) );
+			if ( null === static::$theme ) {
+				$cache_value = static::get_from_persitent_cache( $cache_key );
+				if ( $cache_value ) {
+					static::$theme = $cache_value;
+				}
+			}
+		}
+
+		$has_theme_changed          = false;
+		$has_same_registered_blocks = $can_use_cached ? static::has_same_registered_blocks( 'theme', true ) : static::has_same_registered_blocks( 'theme' );
```

Comment:
This could be simpler

```suggestion
		$has_same_registered_blocks = static::has_same_registered_blocks( 'theme', $can_use_cached );
```